server/server.py

```diff
@@ -3,48 +3,70 @@
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
 import os
+import subprocess
 from pathlib import Path
+from urllib.parse import urlparse
 
-from flask import Flask, request
+from flask import Flask, request, send_from_directory
 from flask.json import jsonify
 from werkzeug.utils import secure_filename
 
 
 URLS = []
-ALLOWED_EXTENSIONS = set(["txt", "pdf", "png", "jpg", "jpeg", "gif", "json"])
+ALLOWED_EXTENSIONS = set(["html"])
+
+path = Path("files")
+path.mkdir(exist_ok=True)
 
-Path("files").mkdir(exist_ok=True)
 app = Flask("klaatu_server")
 app.secret_key = "secret key"
-app.config["UPLOAD_FOLDER"] = "files"
+app.config["UPLOAD_FOLDER"] = path.absolute()
 app.config["MAX_CONTENT_LENGTH"] = 16 * 1024 * 1024
+app.config["SERVER_TYPE"] = os.getenv("KLAATU_SERVER_TYPE", "client")
 
 
 def allowed_file(filename):
     return "." in filename and filename.rsplit(".", 1)[1].lower() in ALLOWED_EXTENSIONS
 
 
-@app.route("/test_results", methods=["POST"])
+@app.route("/test_results", methods=["POST", "GET"])
 def test_results():
+    filename = "index.html"
     request_file = request.files["file"]
-    if request_file and allowed_file(request_file.filename):
-        filename = secure_filename(request_file.filename)
-        request_file.save(os.path.join(app.config["UPLOAD_FOLDER"], filename))
-        resp = jsonify({"message": "File successfully uploaded"})
-        resp.status_code = 201
+    if request.method == "POST":
+        if request_file and allowed_file(request_file.filename):
+            filename = secure_filename(request_file.filename)
+            request_file.save(os.path.join(app.config["UPLOAD_FOLDER"], filename))
+            resp = jsonify({"message": "File successfully uploaded"})
+            resp.status_code = 201
+    elif request.method == "GET":
+        item = send_from_directory(os.path.join(app.config["UPLOAD_FOLDER"]), filename)
```

Comment:
Nit: not sure if you want to move the `filename` declaration from L34 to L42.5 (or hardcode index.html here).

From what I can gather from https://tedboy.github.io/flask/generated/flask.send_from_directory.html, we're just serving the local index.html here for GET requests?
I don't see an index.html in this PR, so I assume it's something generated by cucumber or some other pipeline.

Reply:
That variable needs to be removed.